ansible/roles/elasticsearch/templates/create_es_snapshot.py.j2

```diff
@@ -18,16 +23,31 @@ def create_snapshot(client):
 
 def delete_old_snapshots(client):
     snaps = client.get('{{ es_repository_name }}', '_all')
-    three_days_ago = (datetime.utcnow() - timedelta(days=3)).isoformat()
-    old_snapshots = [s['snapshot'] for s in snaps['snapshots'] if s['start_time'] < three_days_ago]
+    number_of_days_ago = (datetime.utcnow() - timedelta(days={{elastcsearch_backup_days|default(3)}} )).isoformat()
+    ld_snapshots = [s['snapshot'] for s in snaps['snapshots'] if s['start_time'] < number_of_days_ago]
     for snap in old_snapshots:
         client.delete('{{ es_repository_name }}', snap)
 
 
 def main():
     client = get_snapshot_client()
+# Create Today's Snapshot
     create_snapshot(client)
-    delete_old_snapshots(client)
```

Comment:
Since snapshot completion takes some Time this delete call fails. So added a check to see if snapshot creation completed successfully or not.